curriculum/challenges/english/07-scientific-computing-with-python/learn-classes-and-objects-by-building-a-sudoku-solver/6606b224a69a293f98f8db8f.md

```diff
@@ -0,0 +1,55 @@
+---
+id: 6606b224a69a293f98f8db8f
+title: Step 17
+challengeType: 20
+dashedName: step-17
+---
+
+# --description--
+
+The `.index()` method raises a `ValueError` exception when the value is not found. To avoid stopping the program execution, you will nest this line of code inside a `try` block. The `try` statement is used to wrap some code that might raise an exception. The `except` clause instead, provides alternative code to run in case the code inside the `try` block has raised an exception:
```

Comment:
```suggestion
The `.index()` method raises a `ValueError` exception when the value is not found. To prevent the program from halting execution, you'll nest this line of code inside a `try` block. The `try` statement is used to encapsulate code that might raise an exception. The `except` clause, on the other hand, offers alternative code to execute if an exception occurs:
```